src/auth/baseexternalclient.ts

```diff
@@ -708,4 +714,77 @@ export abstract class BaseExternalAccountClient extends AuthClient {
   protected getTokenUrl(): string {
     return this.tokenUrl;
   }
+
+  /**
+   * Returns the workforce identity pool-id if it is determinable
+   * from the audience resource name.
+   * @param audience The STS audience used to determine the pool-id.
+   * @return The pool-id associated with the workforce identity pool, if
+   *   this can be determined from the STS audience field. Otherwise, null is
+   *   returned.
+   */
+  #getWorkForcePoolId(audience: string): string | null {
+    // STS audience pattern:
+    // .../workforcePools/$WORKFORCE_POOL_ID/providers/...
+    return (
+      audience.match(/\/workforcePools\/(?<workforcePool>[^/]+)\/providers\//)
```

Comment:
To make the client more robust, consider adding a check for the audience field in the constructor. This ensures we "fail fast" and propagate a clear error message if a developer tries to instantiate an external client without a required audience. It would prevent a more cryptic TypeError from happening later when audience.match() is called.